### DISCUSSION TITLE
Current Board Rev (2024-06-18) - Programming headers?

### DISCUSSION THREAD

**Discussion by dwareb** [revk/xESP32-Faikout#364] [General]
So, I just installed 5 Daikin mini split heads and was thinking about installing these. I live in the US and I'm very much a tinkerer so I wanted to see if I could get the boards produced myself, which this project has clearly made it really easy to do. Also, since I need 5, and will probably have a few extra made so I can experiment with them, I think I'd rather have a batch made then figure out tricking amazon to allowing me to buy them pre-made. I'm trying to make sure I have all my ducks in a row before I go and order some boards made.

Anyway, the latest board rev doesn't seem to have any programming headers like in the images for older versions. Specifically I was hoping to just use the direct USB headers. The Faikin+ board does have the headers, but it also seems to have 2x S21 headers, and I'm having trouble finding any documentation about what that version of the board is intended for. So I'm a little hesitant to just have those made without knowing what those extra headers are for, or if there are special considerations with that version for programming the board?

Current rev back, no headers:
![faikinback](https://github.com/revk/ESP32-Faikin/assets/53237662/aa91777b-1044-4409-be34-a44f164c5ee9)

Faikin+ back, with headers:
![faikinplusback](https://github.com/revk/ESP32-Faikin/assets/53237662/71ae5408-6c18-4eb3-aea7-de865808ae5f)

Does anyone know what my best move here is? Should I just have the Faikin+ boards produced? Should I scrub through old board releases on the repo and produce one of them? If the older board rev is the best solution, does anyone know which rev or commit I'd be best off rolling it back to?

**Comment by revk** [comment]
Current design has a tag-connect header for the usb - not yet tested in real life.

**Reply by dwareb** [reply]
Oh! I was wondering if those pads were related. Hmm not sure I want to invest in a programmer for that, but I can see how it could speed the programming of a large number of them. I appreciate the info!

**Reply by revk** [reply]
It is possible to solider to them to get USB if needed.

**Reply by dwareb** [reply]
Ah, I didn't think to check for a direct USB pinout. Looks like there is also a direct USB cable that can be purchased as well.

https://www.tag-connect.com/product/tc2030-usb-nl

Thanks again

**Reply by revk** [reply]
Ah useful, I need to check the pin out, as I did not know they did a cable and so picked pins that happen to be convenient. So that cable may not work until I make a change.

**Reply by azerioxal** [reply]
> Current design has a tag-connect header for the usb - not yet tested in real life.

Is there a revision that has been tested and confirmed to work? This revision was in the production folder, so I thought it was good. 
The batch I got from JLCPCB and just got a chance to work on don't seem to work however.

I couldn't get FTDI to work and bought one of the TC2030-USB-NL cables, which also didn't work.
If the pinout doesn't match the TC2030-USB-NL, can you advise the correct pinout so I can try soldering some leads for USB. 
I'd love to salvage these boards if I can... Hopefully I won't need to waste $100.

**Reply by dwareb** [reply]
@azerioxal Out of curiosity what kind of problem are you having? I ended up going through the old commits and having an older rev from earlier in the year produced, as the old rev looked easier to DIY a connector for, than the Tag-Connect one. 

I ran into an issue though where they rebooted / disconnected constantly, but was able to figure out a work around.

**Reply by revk** [reply]
There was, briefly, for a few days, a version with a TC2030 but not their standard USB pin out. I changed this as soon as I realised there was a standard pin out.

You should be able to trace what pins are what though.

**Comment by revk** [comment]
OK latest is TC2030-USB-NL cable compatible (assuming cut down pins work).
